### ISSUE TITLE
Edit Page:  React Single Page App edit more

### ISSUE THREAD

**Issue by fmontes** [dotCMS/core#15756]
# TL;DR
With React Server Side Rendering build a page with container and contentlets `data-attr`. 

# What we're building
1. Normal React website with DotCMS page API ([using create-react-app](https://github.com/facebook/create-react-app))
2. Node server to server side rendering the React pages and send the HTML back to DotCMS Edit Mode
3. Update Edit mode (in core-web) to make it work with regular and React pages

## How SPA edit mode works

Repo: [https://github.com/dotCMS/dotcms-spa-edit-mode](https://github.com/dotCMS/dotcms-spa-edit-mode)

In the most basic, our edit mode only needs a string HTML with special `data-attrs` to work. Right now we get that HTML from the render page endpoint, but when the page is SPA we don't have that, so to get it we do:

### DotCMS
1. We create a [DotCMS plugin](https://github.com/dotCMS/core/files/2742504/plugin.zip) when you install it (`Dev Tools > Plugin > Upload`) it will add a new field to the `Sites` called: "Proxy Url for Edit Mode" in that field we add the url for the Node server.
2. When we add the "Proxy Url for Edit Mode" and go to edit a page, DotCMS will intercept the normal request to the page api and do a POST request to the Node Server with the page object in the body

### Node Server
1. When the Node Server receive the POST request with the page data it will Server Side Render the react page with the special `data-attr` and will response with the string of HTML that our Angular Edit Mode needs.

![edit-mode-spa](https://user-images.githubusercontent.com/751424/50642001-7a39d200-0f2f-11e9-8961-ef5b94c8ab1a.jpg)

## Special HTML data-attr
We use this `data-attr` to edit the page in Angular, with all this information we're able to implement edit contentlet using the `data-dot-inode` or D&D using `data-max-contentlets`, `data-dot-accept` and some others.

### Container
```
<div 
    data-dot-object="container"
    data-dot-inode="e58e92b3-7135-461b-b56b-04ff143a389b" 
    data-dot-identifier="56bd55ea-b04b-480d-9e37-5d6f9217dcc3"
    data-dot-uuid="1"
    data-max-contentlets="25"
    data-dot-accept-types="Document,Blog,webPageContent,Products,News,Media,calendarEvent,Location,WIDGET,FORM"
    data-dot-can-add="CONTENT,WIDGET,FORM"
></div>
```

### Contentlet
```
<div
    data-dot-object="contentlet"
    data-dot-inode="25105f18-4d1a-4be6-a78a-c42ee4c00d61"
    data-dot-identifier="5aef0c62-b7d6-4805-9e7c-77a67f4822f3"
    data-dot-type="calendarEvent"
    data-dot-basetype="CONTENT"
    data-dot-lang="1"
    data-dot-title="Technology Job Fair"
    data-dot-can-edit="true"
    data-dot-content-type-id="f6259cc9-5d78-453e-8167-efd7b72b2e96" 
    data-dot-has-page-lang-version="true"
></div>
```

## Template
The bootstrap template we use:
[unapp.zip](https://github.com/dotCMS/core/files/2817839/unapp.zip)
